fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetaStoreCache.java

```diff
@@ -1031,54 +976,90 @@ public static class HiveFileStatus {
-                    singleUidToColumnRangeMap == null ? null : Maps.newHashMap(singleUidToColumnRangeMap));
-            if (singleColumnRangeMap != null) {
-                RangeMap<ColumnBound, UniqueId> copySingleColumnRangeMap = TreeRangeMap.create();
-                copySingleColumnRangeMap.putAll(singleColumnRangeMap);
-                copy.setSingleColumnRangeMap(copySingleColumnRangeMap);
-            }
+            // sortedPartitionRanges is not copied here, caller should call rebuildSortedPartitionRanges()
+            // after modifying partition data
             return copy;
         }
+
+        /**
+         * Rebuild sorted partition ranges after incremental updates.
+         * Should be called after add/drop partitions.
+         */
+        public void rebuildSortedPartitionRanges() {
+            this.sortedPartitionRanges = buildSortedPartitionRanges();
+        }
+
+        /**
+         * Get sorted partition ranges for binary search filtering.
+         */
+        public Optional<SortedPartitionRanges<String>> getSortedPartitionRanges() {
+            return Optional.ofNullable(sortedPartitionRanges);
+        }
+
+        private SortedPartitionRanges<String> buildSortedPartitionRanges() {
+            if (partitionNameToIdMap == null || partitionNameToIdMap.isEmpty()) {
+                return null;
+            }
+
+            BiMap<Long, String> idToName = partitionNameToIdMap.inverse();
+            List<PartitionItemAndRange<String>> sortedRanges = Lists.newArrayListWithCapacity(
+                    idToPartitionItem.size());
+            List<PartitionItemAndId<String>> defaultPartitions = Lists.newArrayList();
+
+            for (Map.Entry<Long, PartitionItem> entry : idToPartitionItem.entrySet()) {
+                PartitionItem partitionItem = entry.getValue();
+                String partitionName = idToName.get(entry.getKey());
+                if (!partitionItem.isDefaultPartition()) {
+                    List<Range<MultiColumnBound>> ranges = PartitionItemToRange.toRanges(partitionItem);
+                    for (Range<MultiColumnBound> range : ranges) {
+                        sortedRanges.add(new PartitionItemAndRange<>(partitionName, partitionItem, range));
+                    }
+                } else {
+                    defaultPartitions.add(new PartitionItemAndId<>(partitionName, partitionItem));
+                }
+            }
+
+            // Sort by range bounds
+            sortedRanges.sort((o1, o2) -> {
+                Range<MultiColumnBound> span1 = o1.range;
+                Range<MultiColumnBound> span2 = o2.range;
+                int result = span1.lowerEndpoint().compareTo(span2.lowerEndpoint());
+                if (result != 0) {
+                    return result;
+                }
+                return span1.upperEndpoint().compareTo(span2.upperEndpoint());
+            });
+
+            return new SortedPartitionRanges<>(sortedRanges, defaultPartitions);
+        }
```

Comment:
I think you should extract code from `NereidsSortedPartitionsCacheManager.loadCache` and reuse the same utility function